### PR TITLE
Advanced flask homework

### PR DESCRIPTION
**Pull request by sudakevych-andrii**

### CODE REVIEW

File: advanced_flask/supermarkets/main.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        return 'no_photo.png'
+
+
+def save_data():
+    if request.method == 'POST':
+        supermarkets_list = get_data(json_file_path)
+        if len(supermarkets_list):
+            id = str(int(supermarkets_list[len(supermarkets_list) - 1]['id']) + 1)
+        else:
+            id = "1"
```
Comment (dimigor) [inline advanced_flask/supermarkets/main.py]: hardcode

File: advanced_flask/utils.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import json
+
+
+def get_data(file):
+    with open(file) as file:
```
Comment (dimigor) [inline advanced_flask/utils.py]: oh my... 
Do not use python reserved words as variables

Comment (dimigor) [reply advanced_flask/utils.py]: you need to catch exception like file not found or file contains bad data

Comment (sudakevych-andrii) [reply advanced_flask/utils.py]: Done

File: advanced_flask/products/template/product.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+{% block main %}
+    <h1>{{ name }}</h1>
+    <img src="{{ url_for('products.static', filename=img_name) }}" alt="{{ name }}">
+    <div><strong>Price:</strong> {{ price }}</div>
+    <div>
+        <div><strong>Description:</strong></div>
+        <div>{{ description }}</div>
+    </div>
+    <a href="{{url_for('products.get_products')}}">All products</a>
+{% endblock %}
```
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/products/template/product.html]: Done

File: advanced_flask/products/main.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        return image.filename
+    else:
+        return 'no_photo.png'
+
+
+def save_data():
+    if request.method == 'POST':
+        products_list = get_data(json_file_path)
+        if len(products_list):
+            id = str(int(products_list[len(products_list) - 1]['id']) + 1)
```
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/products/main.py]: Done

File: advanced_flask/products/main.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+from wtforms import StringField, SubmitField, TextAreaField, FileField
+from wtforms.validators import DataRequired
+from advanced_flask.utils import get_data, add_data
+
+
+products = Blueprint('products', __name__,
+                     template_folder='template',
+                     static_folder='static',
+                     static_url_path='/products/static')
+json_file_path = 'products/products.json'
```
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/products/main.py]: Done

File: advanced_flask/supermarkets/main.py (hunk 1 above)
Comment (sudakevych-andrii) [reply advanced_flask/supermarkets/main.py]: Done

File: advanced_flask/supermarkets/main.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+            session[id + supermarket['name']] = 'visited'
+            return render_template(
+                'supermarket.html',
+                name=supermarket['name'],
+                location=supermarket['location'],
+                img_name=supermarket['img_name'],
+            )
+
+
+def upload_image():
```
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/supermarkets/main.py]: Done

File: advanced_flask/products/template/product.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/products/template/product.html]: Done

File: advanced_flask/products/main.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/products/main.py]: Done

File: advanced_flask/products/main.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/products/main.py]: Done

File: advanced_flask/products/main.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    return render_template('all_products.html', products_list=products_list)
+
+
+@products.route('/product/<id>')
+def get_product(id):
+    products_list = get_data(json_file_path)
+    for product in products_list:
+        if id == product['id']:
+            session[id + product['name']] = 'visited'
+            return render_template(
```
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/products/main.py]: Done

File: advanced_flask/products/main.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/products/main.py]: Done

File: advanced_flask/products/template/product.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/products/template/product.html]: Done

File: advanced_flask/utils.py (hunk 2 above)
Comment (sudakevych-andrii) [reply advanced_flask/utils.py]: Done

File: advanced_flask/supermarkets/main.py (hunk 1 above)
Comment (sudakevych-andrii) [reply advanced_flask/supermarkets/main.py]: Done

File: advanced_flask/supermarkets/main.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (sudakevych-andrii) [reply advanced_flask/supermarkets/main.py]: Done

File: advanced_flask/products/main.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+                'product.html',
+                name=product['name'],
+                img_name=product['img_name'],
+                price=product['price'],
+                description=product['description']
+            )
+    else:
+        return render_template(url_for('error.html'))
```
Comment (dimigor) [inline advanced_flask/products/main.py]: maybe`url_for('get_error_page')`?

File: advanced_flask/utils.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        image = request_files
+        image.save(os.path.join('products/static', image.filename))
+        return image.filename
+    else:
+        return 'no_photo.png'
+
+
+def save_data(data_file, data):
+    items_list = get_data(data_file)
+    data = data
```
Comment (dimigor) [inline advanced_flask/utils.py]: WTF?

**Pull request merged**